### PR TITLE
cli+build: handle -ObjC flag and route it to MachO linker

### LINKED ISSUE #18784: macho: undefined ObjC symbols when linking ghostty's tests binary
### Zig Version

0.12.0-dev.2542+5a3ae38f3

### Steps to Reproduce and Observed Behavior

As you reported to me by @mitchellh on Discord.

On a macOS host:

```
$ cd ghostty
$ zig build test
...
error: undefined symbol: _OBJC_CLASS_$_CAMetalLayer
    note: referenced by /Users/mitchellh/code/go/src/github.com/mitchellh/ghostty/zig-cache/o/5c47091794c06fdba77eb67932de6361/libcimgui.a(/Users/mitchellh/code/go/src/github.com/mitchellh/ghostty/zig-cache/o/1f454eebdb8632bd389be9e14b9fb977/imgui_impl_metal.o):_OBJC_CLASSLIST_REFERENCES_$_.285
error: undefined symbol: _OBJC_CLASS_$_GCController
    note: referenced by /Users/mitchellh/code/go/src/github.com/mitchellh/ghostty/zig-cache/o/5c47091794c06fdba77eb67932de6361/libcimgui.a(/Users/mitchellh/code/go/src/github.com/mitchellh/ghostty/zig-cache/o/1af7e7c651b709f7ae142f4db1e5b6e7/imgui_impl_osx.o):_OBJC_CLASSLIST_REFERENCES_$_.177
...
```

### Expected Behavior

```
$ cd ghostty
$ zig build test
All tests run fine.
```

### PR DESCRIPTION
**Pull request by kubkon**
Fixes #18784

cc @mitchellh

### CODE REVIEW

**Comment by andrewrk** [comment]
I'm confused, as far as I can tell the only actual change to the linker is fixing a UAF?

File: src/link/MachO.zig [hunk 1]
```diff
@@ -217,6 +217,7 @@ pub fn createEmpty(
         .undefined_treatment = if (allow_shlib_undefined) .dynamic_lookup else .@"error",
         .lib_dirs = options.lib_dirs,
         .framework_dirs = options.framework_dirs,
+        .force_load_objc = options.force_load_objc,
```
Comment (kubkon) [inline src/link/MachO.zig:220]: @andrewrk this is the meaningful change. The machinery to implement `-ObjC` was done by me when upstreaming the revamped linker a few weeks back already - however I didn't plug in parsing of `-ObjC` into the CLI.

Comment (andrewrk) [reply src/link/MachO.zig:220]: ahhh! you already did all the work and simply didn't hook it up yet, I see :)

Comment (kubkon) [reply src/link/MachO.zig:220]: Correct!

**Pull request merged**